src/utils.js

```diff
@@ -163,6 +152,93 @@ function accumulateResults(url, options, property, soFar, totalLimit, token) {
     });
 }
 
+async function yieldResults(url, options) {
+    let {
+        signal,
+        responseKeyName,
+        totalRequestSize,
+        yieldRequestProcessor,
+        formatter
+    } = options;
+
+    if (aborted(signal)) {
+        return [];
+    }
+
+    const yieldRequestHistory = getHistoryYieldRequest(signal);
+    let count = 0;
+    let stop = false;
+    let newUrl = url;
+    let token;
+    let result;
+    let data;
+    let formattedData;
+
+    while (!stop) {
+        result = await yieldRequestHistory.next(newUrl).value;
+        data = result[responseKeyName];
+
+        if (data) {
+            count += data.length;
+            token = result.continuationToken;
+            formattedData = formatter(data);
+
+            if (token) {
+                yieldRequestProcessor(formattedData);
+                newUrl = addTokenToUrl(url, token);
+                yieldRequestHistory.next();
+            } else {
+                return formattedData;
```

Comment:
I wonder if we need to call `yieldRequestProcessor` once here as well? I am thinking that the way this would be used by client code is **either** with the yield callback function **or** with the promise, but not both?

Reply:
Yeah that's a good point. Makes more sense to be consistent. I'll update!